### DISCUSSION TITLE
When does the test runner run compared to the other tasks?

### DISCUSSION THREAD

**Discussion by MRFaria** [tcunit/TcUnit#223] [Q&A] [answered]
Hi all, I am learning to use this tcunit framework but couldn't find the documentation explaining exactly how it works.
I still get a bit confused about all the PLC cyclic code, and wonder how you can guarantee state of the code when everything could be running at the same time and you can't guarantee what else could be affecting the state of your code.

**Comment by sagatowski** [comment, marked as answer]
What do you mean by "everything could be running at the same time"? Are you referring to different tests?
Tests should always be independent of each other, so by definition they will not impact each other if written in this manner.

If you want to run tests sequentially, that is possible too, but then they should still be independent of each other.
See more here:
https://tcunit.org/#/faq?id=_7-is-it-possible-to-run-test-suites-andor-tests-in-a-sequence
